### PR TITLE
upgrade gallery to 0.74

### PR DESCRIPTION
**Pull request by TatianaKapos**
## Description
upgrade Gallery to 0.74

### CODE REVIEW

**Review by chrisglein — APPROVED** [review]
There were more changes in the previous PR: https://github.com/microsoft/react-native-gallery/pull/428#issuecomment-2087214495
Do we not need those extra version bumps?

But as is, this looks fine. Same changes to the app template as far as I can tell.

**Comment by TatianaKapos** [comment]
> There were more changes in the previous PR: [#428 (comment)](https://github.com/microsoft/react-native-gallery/pull/428#issuecomment-2087214495) Do we not need those extra version bumps?
> 
> But as is, this looks fine. Same changes to the app template as far as I can tell.

Ah! Yes some of those package bumps were me trying to get babel to work before I realized we weren't using the correct preset. I'll add back the ones that were upgraded in 0.74 though :)

**Comment by chrisglein** [comment]
CI continues to fail on this:
```
× Building Solution
- Build failed with message 3:34>D:\a\_work\1\s\node_modules\react-native-windows\codegen\NativePlatformConstantsWinSpec.g.h(75,39): error C3615: constexpr function 'winrt::Microsoft::ReactNative::TurboModuleSpec::CheckMethodsHelper' cannot result in a constant expression (compiling source file D:\a\_work\1\s\node_modules\react-native-windows\Microsoft.ReactNative\Modules\PlatformConstantsWinModule.cpp) [D:\a\_work\1\s\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj]. Check your build configuration.
× Build failed with message 3:34>D:\a\_work\1\s\node_modules\react-native-windows\codegen\NativePlatformConstantsWinSpec.g.h(75,39): error C3615: constexpr function 'winrt::Microsoft::ReactNative::TurboModuleSpec::CheckMethodsHelper' cannot result in a constant expression (compiling source file D:\a\_work\1\s\node_modules\react-native-windows\Microsoft.ReactNative\Modules\PlatformConstantsWinModule.cpp) [D:\a\_work\1\s\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj]. Check your build configuration.
- It is possible your installation is missing required software dependencies. Dependencies can be automatically installed by running D:\a\_work\1\s\node_modules\react-native-windows\scripts\rnw-dependencies.ps1 from an elevated PowerShell prompt.
For more information, go to http://aka.ms/rnw-deps
× It is possible your installation is missing required software dependencies. Dependencies can be automatically installed by running D:\a\_work\1\s\node_modules\react-native-windows\scripts\rnw-dependencies.ps1 from an elevated PowerShell prompt.
For more information, go to http://aka.ms/rnw-deps
Command failed with error MSBuildError: 3:34>D:\a\_work\1\s\node_modules\react-native-windows\codegen\NativePlatformConstantsWinSpec.g.h(75,39): error C3615: constexpr function 'winrt::Microsoft::ReactNative::TurboModuleSpec::CheckMethodsHelper' cannot result in a constant expression (compiling source file D:\a\_work\1\s\node_modules\react-native-windows\Microsoft.ReactNative\Modules\PlatformConstantsWinModule.cpp) [D:\a\_work\1\s\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj]
##[error]Cmd.exe exited with code '2302'.
```
[source](https://dev.azure.com/ms/react-native-gallery/_build/results?buildId=570597&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=b24cf96b-e476-5cd6-97ae-776d6935d22f&l=6539)

This doesn't repro for me when building the branch locally.

Failure first occurs from building these files:
- Timing.cpp
- NativeModulesProvider.cpp

(note: RNW files, not gallery files)

There doesn't seem to be any CI errors when building these samples on 0.74 though: https://github.com/microsoft/react-native-windows-samples/pull/937
@tatianakapos Are there differences in the CI between the repos? Or differences between construction of the apps?
If we want a third example, maybe you can help me onboard better CI validation of Artificial Chat so this PR would give us signal? https://github.com/chrisglein/artificial-chat/pull/145

**Comment by TatianaKapos** [comment]
@chrisglein Interestingly, this is the same error message from [this issue](https://github.com/microsoft/react-native-windows/issues/7748) on our repo which has to do with switching to C++20, I'm not sure why only the Gallery CI would be hitting this.

A big comparison between Gallery and our samples repo is that Gallery uses ADO while Samples uses github workflows. Gallery also uses yarm instead of yarn which I want to try switching to see if that helps!

More than happy to add CI to Artificial Chat though, it would be helpful to have multiple apps running the same build structure.

File: windows/rngallery.sln [hunk 1]
```diff
@@ -53,6 +53,8 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "RNCConfig", "..\node_module
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "RNDeviceInfoCPP", "..\node_modules\react-native-device-info\windows\RNDeviceInfoCPP\RNDeviceInfoCPP.vcxproj", "{3E3931F2-4735-4417-8CB0-33668A7314D6}"
 EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "RNLinearGradient", "..\node_modules\react-native-linear-gradient\windows\RNLinearGradient\RNLinearGradient\RNLinearGradient.vcxproj", "{193DF456-3F97-4CC1-A7D4-809857CC064E}"
+EndProject
```
Comment (chrisglein) [inline windows/rngallery.sln:57]: Mis-merge duplicate add
```suggestion
Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "RNLinearGradient", "..\node_modules\react-native-linear-gradient\windows\RNLinearGradient\RNLinearGradient\RNLinearGradient.vcxproj", "{193DF456-3F97-4CC1-A7D4-809857CC064E}"
EndProject
```
(it's already there on line 74)

File: windows/rngallery.sln [hunk 2]
```diff
@@ -283,6 +283,18 @@ Global
...
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Debug|x64.ActiveCfg = Debug|x64
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Debug|x64.Build.0 = Debug|x64
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Debug|x86.ActiveCfg = Debug|Win32
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Debug|x86.Build.0 = Debug|Win32
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Release|ARM64.ActiveCfg = Release|ARM64
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Release|ARM64.Build.0 = Release|ARM64
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Release|x64.ActiveCfg = Release|x64
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Release|x64.Build.0 = Release|x64
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Release|x86.ActiveCfg = Release|Win32
+		{193DF456-3F97-4CC1-A7D4-809857CC064E}.Release|x86.Build.0 = Release|Win32
```
Comment (chrisglein) [inline windows/rngallery.sln:297]: Not sure I understand why this part is here
```suggestion
```
The guid `193DF456-3F97-4CC1-A7D4-809857CC064E` points to the linear gradient. And appears to be in duplicate. Safe to remove?

File: README.md [hunk 3]
```diff
@@ -2,7 +2,8 @@
 
 Welcome to React Native Gallery. A React Native Windows application which displays the range of React Native components with Windows support. React Native Gallery is currently live in the Microsoft Store! Follow the link to [download](https://www.microsoft.com/en-us/p/react-native-gallery/9npg0b292h4r).
 
-![gallery_welcome](https://user-images.githubusercontent.com/34109996/115446025-777c4d00-a1cb-11eb-8a8f-c1758613c09f.png)
+![gallery_welcome](https://github.com/TatianaKapos/react-native-gallery/assets/42554868/6f78c5fd-3e20-4eaf-aa04-34d6289cfb0c)
```
Comment (chrisglein) [inline README.md:5]: No worse than the link that was there before. But do we make more stable links in the .github folder or something?

Comment (TatianaKapos) [reply README.md]: I just added it to the assets folder!

**Pull request merged**